cmd/object-handlers.go

```diff
@@ -806,28 +837,19 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 		return
 	}
 
-	doneCh := make(chan struct{})
-
-	// Signal that completeMultipartUpload is over via doneCh
+	// Signal that joining parts is over is over via doneCh
 	go func(doneCh chan<- struct{}) {
-		md5Sum, err = objectAPI.CompleteMultipartUpload(bucket, object, uploadID, completeParts)
+		md5Sum, err = joinParts()
 		doneCh <- struct{}{}
 	}(doneCh)
 
 	sendWhiteSpaceChars(w, doneCh)
```

Comment:
So I'll keep the CompleteMultipart API signature as it was and remove the sending of white space characters and running complete multipart in a new routine.